### PR TITLE
remove explicit iter loops

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

**Comment by panekj** [comment]
Does this have any meaningful impact such as improved performance or is it only a cosmetic change?

**Comment by danieleades** [comment]
> Does this have any meaningful impact such as improved performance or is it only a cosmetic change?

it's cosmetic. Additionally, there's inherent value in suppressing/addressing things clippy complains about because it makes the linting less noisy, more likely to be used, and therefore more able to generate actual value. I would personally argue that's valuable enough in it's own right, though compared to #1871, this clippy 'noise' is at `clippy::pedantic` level, rather than `clippy::all`, so it's a slightly harder case to make.

**Comment by JustForFun88** [comment]
It seems to me that the changes only increase the compilation time, since now the compiler still has to infer the call to these `iter` and `iter_mut` functions itself.

**Comment by danieleades** [comment]
> It seems to me that the changes only increase the compilation time, since now the compiler still has to infer the call to these `iter` and `iter_mut` functions itself.

Have you done some measurement that shows this? I would find an increase in compilation time here to be *very* surprising

**Comment by JustForFun88** [comment]
> Have you done some measurement that shows this? I would find an increase in compilation time here to be _very_ surprising

Yes, my mistake, you are right 😊. Looks like the `for` loop always results in an implicit call to `IntoIterator`, so both constructs are compile-time equivalent.

File: lapce-core/src/syntax/util.rs [hunk 1]
```diff
@@ -123,12 +123,12 @@ pub fn has_unmatched_pair(line: &str) -> bool {
             }
         }
     }
-    for (_, pair_count) in count.iter() {
+    for (_, pair_count) in &count {
```
Comment (Arjun31415) [inline lapce-core/src/syntax/util.rs:126]: I believe you need to use `count.values()`

**Review by Arjun31415 — APPROVED** [review]

**Review by panekj — CHANGES_REQUESTED** [review]
waiting for next clippy version

**Comment by panekj** [comment]
Closing since those lints don't seem to be default anytime soon

**Pull request closed without merging**